src/thunderbots/software/ai/primitive/stop_primitive.cpp

```diff
@@ -0,0 +1,42 @@
+#include "ai/primitive/stop_primitive.h"
+
+const std::string StopPrimitive::PRIMITIVE_NAME = "Stop Primitive";
+
+StopPrimitive::StopPrimitive(unsigned int robot_id, bool coast) : robot_id(robot_id), coast(coast)
+{
+}
+
+StopPrimitive::StopPrimitive(const thunderbots_msgs::Primitive &primitive_msg)
+{
+    validatePrimitiveMessage(primitive_msg, getPrimitiveName());
+
+    robot_id = primitive_msg.robot_id;
+    coast = primitive_msg.extra_bits.at(0);
+}
+
+std::string StopPrimitive::getPrimitiveName() const
+{
+    return PRIMITIVE_NAME;
+}
+
+unsigned int StopPrimitive::getRobotId() const
+{
+    return robot_id;
+}
+
+bool StopPrimitive::coastRobot() const
+{
+    return coast;
+}
+
+std::vector<double> StopPrimitive::getParameters() const
+{
+    return std::vector<double>();
+}
+
+std::vector<bool> StopPrimitive::getExtraBits() const
+{
+    std::vector<bool> parameter = {coast};
+
+    return parameter;
+}
```

Comment:
Github complaining about lack of newline at end of file.

Reply:
bump